solutions/advanced/cses-2187.mdx

```diff
@@ -0,0 +1,120 @@
+---
+id: cses-2187
+source: CSES
+title: Bracket Sequences II
+author: Mihnea Brebenel
+---
+
+## Explanation
+
+First, it would be helpful to know how many open and closed brackets should be in the remaining sequence; let $c$ and $n$
+denote the number of closed brackets and the length of the remaining sequence. Then, the number of possible endings -
+not necessarily valid - is ${n \choose c}$.
+
+Second, not all ${n \choose c}$ endings are valid, so by counting the invalid endings we'll know that the rest must be valid.
+To count the invalid endings, flip the brackets in the suffix beginning from the firt position the sequence becomes invalid,
+resulting in $c+1$ closed brackets, thus the number of invalid endings is ${n \choose c+1}$. The answer is the difference between
```

Comment:
https://en.wikipedia.org/wiki/Catalan_number#Second_proof
honestly this link explains it better

Reply:
Ok should I just add it or remove what I wrote ?

Reply:
second

Reply:
:((

Reply:
Ok I removed it and added the wiki